pulsar-broker/src/main/java/org/apache/pulsar/compaction/PulsarTopicCompactionService.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.compaction;
+
+import static org.apache.pulsar.compaction.CompactedTopicImpl.COMPACT_LEDGER_EMPTY;
+import static org.apache.pulsar.compaction.CompactedTopicImpl.NEWER_THAN_COMPACTED;
+import static org.apache.pulsar.compaction.CompactedTopicImpl.findStartPoint;
+import java.util.Collections;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import org.apache.bookkeeper.client.BookKeeper;
+import org.apache.bookkeeper.mledger.Entry;
+import org.apache.bookkeeper.mledger.Position;
+import org.apache.bookkeeper.mledger.impl.PositionImpl;
+
+
+public class PulsarTopicCompactionService implements TopicCompactionService {
+
+    private final String topic;
+
+    private final CompactedTopicImpl compactedTopic;
+
+    private final Supplier<Compactor> compactorSupplier;
+
+    public PulsarTopicCompactionService(String topic, BookKeeper bookKeeper,
+                                        Supplier<Compactor> compactorSupplier) {
+        this.topic = topic;
+        this.compactedTopic = new CompactedTopicImpl(bookKeeper);
+        this.compactorSupplier = compactorSupplier;
+    }
+
+    @Override
+    public CompletableFuture<Void> compact() {
+        Compactor compactor;
+        try {
+            compactor = compactorSupplier.get();
+        } catch (CompletionException e) {
+            return CompletableFuture.failedFuture(e);
+        }
+        return compactor.compact(topic).thenApply(x -> null);
+    }
+
+    @Override
+    public CompletableFuture<List<Entry>> readCompactedEntries(@Nonnull Position startPosition,
+                                                               int numberOfEntriesToRead) {
+        CompletableFuture<List<Entry>> resultFuture = new CompletableFuture<>();
```

Comment:
1. Should we add an NPE check for `startPosition`?
2. Should we add `numberOfEntriesToRead` 0 value check?